hack/tools/release/notes.go

```diff
@@ -307,8 +309,11 @@ func run() int {
 		}
 	}
 
+	if *releaseCandidate {
+		fmt.Printf("🚨 This is a RELEASE CANDIDATE. Use it only for testing purposes. If you find any bugs, file an [issue](https://github.com/kubernetes-sigs/cluster-api/issues/new).\n")
```

Comment:
Do we need to do this for a beta aswell? Would it be easier to detect this from the structure of the release tag rather than changing the flags depending on the kind of release?

Reply:
Indeed, this would be a good idea. It could also help with other "variable" sections, like the expandable blocks for RC/Beta (with the full commits diff against las minor release).

However, it requires re-designing the input to this command, at least partly. Right now, it takes the base tag/commit you want to use to compute the diff with HEAD (of current branch). This approach would move it to take the "next tag" as the input and it will need to compute the `fromTag` dynamically based on certain rules (which I think we can easily come up with based on our release schedule).

I'll leave that up to y'all, I don't have a hard opinion, but I would be ok merging this as is (we get some value delivered immediately) and then working on this input redesign as one of our next improvement tasks.

Reply:
if we need beta as well, then renaming it to something like more generic, i.e `pre-release version` or similar might fit better

Reply:
I think we can do this easily if we switch to the kubernetes release tool, which I'm taking a look at through #9249

As it is now I don't think there's a straightforward way to make this "smarter" - I can update this PR to use the naming `--pre-release-version` though since I agree that's clearer.